modules/p2/ding_entity_rating/ding_entity_rating.module

```diff
@@ -417,26 +400,38 @@ function ding_entity_rating_remove_from_list($entity, $account) {
   }
 }
 
+function ding_entity_rating_ding_interaction_info() {
+  global $user;
+  return array(
+    'title' => t('Rate materials and get better serendipity'),
+    'description' => t('If the user has not rated any materials yet, ' .
+      'suggest this can be done for instance in the loan history.'),
+    'active' => ding_user_is_provider_user($user),
+    'css_class' => 'complete-info',
+
+  );
+}
+
 /**
- * Implements hook_ding_user_complete().
+ * Implements hook_ding_interaction_view().
  */
-function ding_entity_rating_ding_user_complete() {
-  $complete = FALSE;
-
+function ding_entity_rating_ding_interaction_view() {
+  global $user;
   $user_history_list = ding_list_get_lists(DING_LIST_TYPE_LOAN_HISTORY);
   $history_list = current($user_history_list);
-  global $user;
+
+  $uid = $user->uid;
+  $history_button = isset($history_list) ? "<li>" . l(t("Rate previous loans"), "list/" . $history_list->id) . "</li>" : "";
   
-  return array(
-    'ding_entity_rating' => array(
-      'title' => t('It pays to judge'),
-      'points' => 3,
-      'completed' => $complete,
-      'markup' => theme('ding_entity_rating_ding_user_complete_description', array('history_list' => $history_list, 'account' => $user)),
-    ),
-  );
+  return "<div class='description'>" .
```

Comment:
Why drop the theme function in favor of outputting HTML directly?
If you need to return HTML, you should create a render array with a #theme property, and use drupal_render to uotput the rendered HTML.